src/js/state/Tabs/slice.ts

```diff
@@ -85,7 +85,7 @@ const slice = createSlice({
   },
   extraReducers: (builder) => {
     builder.addMatcher(isTabAction, (s, a) => {
-      const id = a.tabId || s.active
+      const id = a.tabId || a.payload?.tabId || s.active
```

Comment:
This fixes the second problem you found. Where the results and the stats are updating the wrong tab.